stack-graphs/src/partial.rs

```diff
@@ -2679,18 +2688,23 @@ impl Node {
             Self::PushScopedSymbol(node) => {
                 let symbol = symbol_stack
                     .pop_front(partials)
-                    .expect("missing symbol for push scoped symbol node in postcondition");
-                debug_assert_eq!(symbol.symbol, node.symbol);
+                    .ok_or(PathResolutionError::EmptySymbolStack)?;
+                if symbol.symbol != node.symbol {
+                    return Err(PathResolutionError::IncorrectPoppedSymbol);
+                }
```

Comment:
I like this cleanup — gives this function a more complex signature but allows more use of `?` in a useful way